js/modules/k6/http/request.go

```diff
@@ -73,6 +73,61 @@ func (c *Client) Request(method string, url goja.Value, args ...goja.Value) (*Re
 	return c.responseFromHTTPext(resp), nil
 }
 
+// Request makes an http request of the provided `method` and returns a corresponding response by
+// taking goja.Values as arguments
+func (c *Client) asyncRequest(method string, url goja.Value, args ...goja.Value) (*goja.Promise, error) {
+	state := c.moduleInstance.vu.State()
+	if state == nil {
+		return nil, ErrHTTPForbiddenInInitContext
+	}
+
+	var body interface{}
+	var params goja.Value
+
+	if len(args) > 0 {
+		body = args[0].Export()
+	}
+	if len(args) > 1 {
+		params = args[1]
+	}
+
+	rt := c.moduleInstance.vu.Runtime()
+	req, err := c.parseRequest(method, url, body, params)
+	p, resolve, reject := rt.NewPromise()
+	if err != nil {
+		if state.Options.Throw.Bool {
+			reject(err)
+			return p, nil
+		}
+		state.Logger.WithField("error", err).Warn("Request Failed")
+		r := httpext.NewResponse()
+		r.Error = err.Error()
+		var k6e httpext.K6Error
+		if errors.As(err, &k6e) {
+			r.ErrorCode = int(k6e.Code)
+		}
+		resolve(&Response{Response: r, client: c})
+		return p, nil
```

Comment:
Most of this method is repeated in `Request()`. Can we DRY it a bit?

Calling `MakeRequest()` below might be tricky, but everything above seems like it could go in a reusable function, so that the only difference between `Request()` and `asyncRequest()` is the async part of it. Similar to what you did in `tracing.Client`.

Reply:
In the tracing.Client the only different part was what was returned. 

I kind of don't want to touch `http.request` if possible as I prefer to keep it as unbroken as possible :sweat_smile: 

[2f40547](https://github.com/grafana/k6/pull/2877/commits/2f405474b192e9e13fb3b9a3caee74eb278f17d1) does a lot of the work IMO, but DRYing the rest IMO will just make it way more unreadable than anything else